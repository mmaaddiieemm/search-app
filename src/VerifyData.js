import React from 'react';

// given a dataset and a parameter, return the formatted data as a string
// if the data is valid, otherwise returns 'undefined'. If the data is an array,
// will add some extra characters to make it more readable.
function VerifyData(data, param) {
  // check to make sure the data is valid, if not return 'undefined'
  if (typeof data[param] === 'undefined' || data[param] === null || !data.hasOwnProperty(param) || data === null) {
    return <i>undefined</i>;
  }
  let result = data[param].toString();
  // do a little extra formatting for arrays.
  if (Array.isArray(data[param])) {
    result = `[${result}]`;
  }
  return result;
}

export default VerifyData;
