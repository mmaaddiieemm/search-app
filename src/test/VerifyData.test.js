import React from "react";
import VerifyData from "../VerifyData.js"; 
import renderer from 'react-test-renderer';

let testData = {
    testName: 'notMaddie', 
    _id: '4',
    others: false,
    description: 'emtpy!'
};

test('When an entry is null, return undefined', () => {
    const component = renderer.create(
        <VerifyData data={testData} param={null} />
    ); 
    let tree = component.toJSON(); 
    expect(tree).toMatchSnapshot(); 
}); 
test('When an entry doesnt contain the given param, return undefined', () => {
    const component = renderer.create(
        <VerifyData data={testData} param={'hasNeighbors'} />
    ); 
    let tree = component.toJSON(); 
    expect(tree).toMatchSnapshot(); 
}); 