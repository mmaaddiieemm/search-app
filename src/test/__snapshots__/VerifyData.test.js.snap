// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When an entry doesnt contain the given param, return undefined 1`] = `
<i>
  undefined
</i>
`;

exports[`When an entry is null, return undefined 1`] = `
<i>
  undefined
</i>
`;
